### DISCUSSION TITLE
umi router 如何刷新当前路由

### DISCUSSION THREAD

**Discussion by wxq851685279** [umijs/umi#7984] [General]
使用window.location.reload()  刷新了整个页面了，如何只刷新路由页面

**Comment by qiutian00** [comment]
`history.push`, you can see this page: https://umijs.org/zh-CN/api#history

**Reply by wxq851685279** [reply]
我试了，不行，如果当前路由是 /user/create,  history.push("/user/create") 没反应

**Reply by syzhang914** [reply]
> 我试了，不行，如果当前路由是 /user/create, history.push("/user/create") 没反应

我也是之前升级了umi3有这个问题，后来发现不是没有刷新当前路由，而是之前的写法是从state里取route值赋给了props的location，但其实升级后本身location就会被传到props里，这样赋值反而覆盖了本身的location所以修改后没有反应

**Comment by xierenyuan** [comment]
可以参考https://stackoverflow.com/questions/46820682/how-do-i-reload-a-page-with-react-router  这解决
export const Page = () => {
   const location = useLocation();
   return <PageImpl key={location.key} />
}
或者
const location = useLocation()

useEffect(() => {
  fetchData()
}, [location.key])
